aasemble/django/apps/api/tests.py

```diff
@@ -400,3 +400,20 @@ def test_github_push_event(self, poll_one):
         from .tasks import github_push_event
         github_push_event("https://github.com/eric/project0")
         poll_one.delay.assert_called_with(1)
+
+
+class APIv1MirrorsetTests(APIv1Tests):
+    list_url = '/api/v1/mirror_sets/'
+
+    def test_create_mirrorset_empty_fails_400(self):
+        # TODO: The funny thing is, this scenario succeeds in creating mirrorset
+        # on the prod server returning a status code of 201 CREATED. Check why.
```

Comment:
Wow. It took a while to work this out.

The difference is: `self.client.post(url, {}, format='json')` actually sends an empty json dictionary.

Sending an empty request using curl or httpie or whatever sends *nothing*. Nothing gets turned into a QueryDict. A QueryDict is handled slightly differently than a regular dictionary.